CONTRIBUTING.md

```diff
@@ -15,24 +15,62 @@ We welcome contributions to the codebase. Here are some coding guidelines to fol
-- We do regular minor releases. These are releases that are additive and don't break existing functionality. During a minor release we bump the minor version number: 0.41.24 -> 0.42.24
-- In-between minor releases, we can release bug-fixes by bumping the patch number: 0.34.6 -> 0.34.7
-- If there are any breaking changes, there will be a major release: 5.39.55 -> 6.39.55
-- Not that pre-version 1.x.x, because this is the pre-release stage, breaking functionality may be introduced in minor versions.
+- A Github integration pipeline will run. After all tests pass, build a multiplatform docker image for the `linux/arm64` and `linux/amd64` platforms, pushing it to the
+`marqoai/marqo` repository. Make sure it is tagged with the version number (`marqoai/marqo:0.1.5`)   
+- For releases, please record changes in  `RELEASE.md`. Then create a 
+[github release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) 
+(with a short summary of changes) that links to the changes in `RELEASE.md`.
+- For releases that change the API, update py-marqo to make those changes accessible. 
+- To release a change to py-marqo, bump py-marqo's version (major.minor.patch, see below for more details), run all the tests, build the package and push to it PyPi. py-marqo's verson is independent from Marqo.  
+
+- Then, use the following format for releasing changes:
+  - . # Release x.y.z 
+    - small blurb about release
+  - . ## Breaking changes
+  - . ## Deprecations
+  - . ## Caveats 
+    - adding/bumping of dependencies
+    - lack of platform support (for certain features)
+    - other (non-breaking) caveats to new features/fixes 
+  - . ## New features
+    - Non-breaking changes
+  - . ##  Bug fixes 
+  - . ### Testing
 
 ### Testing
-Notes: 
 
-- If you don't have the exact python version specified in the `tox.ini` file, you can run tox like this: `tox -e py`. This tells `tox` to use whatever python version it finds on your machine. 
-- Be very careful if your environment has access to a Marqo production instance. These tests will arbitrarily modify, create and delete indices on the Marqo instances it uses for testing. 
+WARNING: Be very careful if your environment has access to a Marqo production instance. These tests will arbitrarily modify, create and delete indices on the Marqo instances it uses for testing. 
 
+
+__Tips__
+
+If you don't have the exact python version specified in the `tox.ini` file, you can run tox like this: `tox -e py`. This tells `tox` to use whatever python version it finds on your machine.
 #### Unit tests
-To run unit tests, simple run `tox` in the Marqo home directory
+
+1. Ensure you have marqo-os running:
+```bash
+docker rm -f marqo-os
+docker run -p 9200:9200 -p 9600:9600 -e "discovery.type=single-node" --name marqo-os marqoai/marqo-os:0.0.2-arm
```

Comment:
should we recommend a docker pull first?

Reply:
This will just pull the latest version of marqo-os, which probably isn't as useful as pulling marqo